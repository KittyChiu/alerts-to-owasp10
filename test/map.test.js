const fs = require('fs')
const mapRisksToAlerts = require('../src/map') // replace './map' with the actual path to the module

describe('mapRisksToAlerts', () => {
  it('should create a CSV file with the correct data', () => {
    // Call the function with test data
    mapRisksToAlerts(
      'test/sample_data/owasp10.json',
      'test/sample_data/alerts.json',
      'mapping.csv'
    )

    // Read the created CSV file
    const csvData = fs.readFileSync('mapping.csv', 'utf8')

    // Compare the CSV data with a snapshot
    expect(csvData).toMatchSnapshot()
  })
})
