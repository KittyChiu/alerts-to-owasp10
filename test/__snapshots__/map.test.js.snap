// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapRisksToAlerts should create a CSV file with the correct data 1`] = `
"repo_name,alert_no,risk,cwe_id
webgoat-demo-2,1,A03:2021 – Injection,cwe-079
webgoat-demo-2,2,A03:2021 – Injection,cwe-079
webgoat-demo-2,3,A03:2021 – Injection,cwe-079
webgoat-demo-2,4,A03:2021 – Injection,cwe-079
webgoat-demo-2,5,A03:2021 – Injection,cwe-079
webgoat-demo-2,6,A03:2021 – Injection,cwe-079
webgoat-demo-2,7,A03:2021 – Injection,cwe-079
webgoat-demo-2,8,A03:2021 – Injection,cwe-079
webgoat-demo-2,9,A03:2021 – Injection,cwe-079
webgoat-demo-2,10,A03:2021 – Injection,cwe-020
webgoat-demo-2,11,A03:2021 – Injection,cwe-020
webgoat-demo-2,12,A03:2021 – Injection,cwe-020
webgoat-demo-mb,1,A03:2021 – Injection,cwe-079
webgoat-demo-mb,2,A03:2021 – Injection,cwe-079
webgoat-demo-mb,3,A03:2021 – Injection,cwe-079
webgoat-demo-mb,4,A03:2021 – Injection,cwe-079
webgoat-demo-mb,5,A03:2021 – Injection,cwe-079
webgoat-demo-mb,6,A03:2021 – Injection,cwe-079
webgoat-demo-mb,7,A03:2021 – Injection,cwe-079
webgoat-demo-mb,8,A03:2021 – Injection,cwe-079
webgoat-demo-mb,9,A03:2021 – Injection,cwe-079
webgoat-demo-mb,10,A03:2021 – Injection,cwe-020
webgoat-demo-mb,11,A03:2021 – Injection,cwe-020
"
`;
